api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentStatusController.java

```diff
@@ -0,0 +1,159 @@
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import uk.gov.hmcts.payment.api.configuration.LaunchDarklyFeatureToggler;
+import uk.gov.hmcts.payment.api.dto.*;
+import uk.gov.hmcts.payment.api.dto.mapper.PaymentStatusResponseMapper;
+import uk.gov.hmcts.payment.api.exception.FailureReferenceNotFoundException;
+import uk.gov.hmcts.payment.api.model.Payment;
+import uk.gov.hmcts.payment.api.model.Payment2Repository;
+import uk.gov.hmcts.payment.api.model.PaymentFailures;
+import uk.gov.hmcts.payment.api.service.PaymentStatusUpdateService;
+import uk.gov.hmcts.payment.api.v1.model.exceptions.PaymentNotFoundException;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+
+@RestController
+@Api(tags = {"Payment Status"})
+@SwaggerDefinition(tags = {@Tag(name = "PaymentStatusController", description = "Payment Status REST API")})
+public class PaymentStatusController {
+    private static final Logger LOG = LoggerFactory.getLogger(PaymentStatusController.class);
+
+    @Autowired
+    private PaymentStatusUpdateService paymentStatusUpdateService;
+
+    @Autowired
+    private Payment2Repository paymentRepository;
+
+    @Autowired
+    private PaymentStatusResponseMapper paymentStatusResponseMapper;
+
+    @Autowired
+    private LaunchDarklyFeatureToggler featureToggler;
+
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "successful operation"),
+        @ApiResponse(code = 400, message = "Bad request"),
+        @ApiResponse(code = 404, message = "No Payments available for the given Payment reference"),
+        @ApiResponse(code = 429, message = "Request already received for this failure reference"),
+        @ApiResponse(code = 500, message = "Internal Server Error")
+
+    })
+    @PaymentExternalAPI
+    @PostMapping(path = "/payment-failures/bounced-cheque")
+    public ResponseEntity<String> paymentStatusBouncedCheque(@Valid @RequestBody PaymentStatusBouncedChequeDto paymentStatusBouncedChequeDto){
+
+        boolean psuLockFeature = featureToggler.getBooleanValue("payment-status-update-flag",false);
+        LOG.info("feature toggler enable for  bounced-cheque : {}",psuLockFeature);
+
+        if(psuLockFeature){
+            return new ResponseEntity<>("service unavailable", HttpStatus.SERVICE_UNAVAILABLE);
+        }
+
+        LOG.info("Received payment status request bounced-cheque : {}", paymentStatusBouncedChequeDto);
+        Optional<Payment> payment = paymentRepository.findByReference(paymentStatusBouncedChequeDto.getPaymentReference());
+
+        if(payment.isEmpty()){
+            throw new PaymentNotFoundException("No Payments available for the given Payment reference");
+        }
+
+        Optional<PaymentFailures> paymentFailures = paymentStatusUpdateService.searchFailureReference(paymentStatusBouncedChequeDto.getFailureReference());
```

Comment:
Why are 2 calls being made to Payment Failures table - one for checking the failure reference record and then inserting if not found? why cannot it be done in 1 call, if record already exists then throw an exception.